### PR TITLE
Notepad with tabs: speak when switching between tabs

### LINKED ISSUE #14587: in tabbed notepad when switching between tabs nvda should announce some way to differentiate between tabs
### Steps to reproduce:
download the new tabbed notepad.
now using the menu create a new tab
now switch between tabs with ctrl+tabe
### Actual behavior:
nvda announces blank edition text editor
### Expected behavior:
Before writing what I want, I would like to talk about my discoveries, sorry if it doesn't make sense.
I typed a different word into the first line of text on each tab.
guide example 1
Fernando
guide 2
silva
using object navigation I found the list of tabs and within this list there was each tab named with what was written in the first line of text.
Now I left the first line of text empty in tab 1
in the list of tabs tab 1 appears with the name of untitled
from what i understand if the first line of text is characters this text will be the title of the tab.
If the first line of text is empty, the tab will have an untitled title.
so my suggestion is:
when switching between tabs in notepad in this example by pressing ctrl+tab nvda should announce the title of the tab which will be what is typed in the first line.
But this doesn't work if the first line of the tabs is empty, so I suggest that nvda also announce the position of the tab within the list.
example
guide 1
first line
Fernando
guide 2
first line
empty
guide 3
first line
silva
when switching between tabs nvda would announce:
guide 1 of 3 fernando
guide 2 of 3 untitled
guide 3 of 3 silva
Tab name and tab count could also be announced by command nvda + t to read window name.
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
nvda.exe, NVDA alpha-27590,180c9f2b
#### Windows version:
11 22.623.1095
#### Name and version of other software in use when reproducing the issue:
Notepad.exe, Microsoft.WindowsNotepad 11.2212.33.0

#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

### PR DESCRIPTION
**Pull request by mzanm**
### Link to issue number:
Closes #14587 

### Summary of the issue:
In recent notepad releases on windows 11 insider, notepad has gained the ability to open multiple files in the same window with tabs. When switching between tabs with control tab, NVDA does not the speak the currently selected tab or it has changed at all.

### Description of user facing changes
When CTRL tab is pressed, the new tab name will be spoken, similar to web browsers.

### Description of development approach
Implement behavior for UIA_elementSelected, if the object is a tab and is selected, speak it.

### Testing strategy:
Created multiple tabs in notepad and switched between them.
### Known issues with pull request:
None.

### Change log entries:
Bug fixes
- In windows 11 with newer notepad, switching between tabs will announce tab name and position.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by josephsl** [comment]
Hi,

I see. I understand that this works for speech users. Have you considered the braille experience?

Thanks.

**Comment by mazen428** [comment]
Fixed. Thanks.

**Comment by mazen428** [comment]
I agree a general solution would be better, but some XAML apps already announce tabs in some way, like windows terminal, so adding a general solution will cause double speaking.

**Comment by josephsl** [comment]
Hi,

For Windows Terminal, the name of the tab is part of the focus announcement, hence double speaking when switching between them. This is not the case for tabbed Notepad and File Explorer. This was confirmed with both NVDA and Narrator.

Thanks.

**Comment by josephsl** [comment]
@SeanBudd, may I ask for a code review? I approve this PR but won't hurt for you to make the final decision. Getting this into 2023.1 would be fine, or we can wai until 2023.2 since tabbed Notepad won't make its way to folks using Windows 11 stable releases for some time (give several weeks).

Thanks.

**Review by josephsl — APPROVED** [review]
Thanks for this work - I think this is your first NVDA PR, and congraulations on the job well done.

**Review by michaelDCurran — APPROVED** [review]

File: source/appModules/notepad.py
Comment (seanbudd) [inline source/appModules/notepad.py:27]: Some formatting suggestions for readibility

```suggestion
			and not eventHandler.isPendingEvents("UIA_elementSelected") and controlTypes.State.SELECTED in obj.states
		):
```

File: source/appModules/notepad.py
Comment (seanbudd) [inline source/appModules/notepad.py:31]:
```suggestion
			braille.handler.message(
				braille.getPropertiesBraille(
					name=obj.name,
					role=obj.role,
					states=obj.states,
					positionInfo=obj.positionInfo
				)
			)
```

**Comment by mazen428** [comment]
Waiting for #14622 to be merged.

**Comment by josephsl** [comment]
Hi, you don’t have to wait for my work to be merged – although there could be conflicts due to code positioning, they are independent unless this PR uses parts of the linked PR. Thanks.

**Comment by seanbudd** [comment]
@mazen428 - that PR has been merged now, thanks for waiting

**Review by seanbudd — COMMENTED** [review]
Hi @mazen428 - just some minor issues remaining

**Review by seanbudd — APPROVED** [review]

**Pull request merged**
